### ISSUE TITLE
InvalidOperationException Collection was modified

### ISSUE THREAD

**Issue by IlyaKvant** [FirstGearGames/FishySteamworks#9]
Unity: 2021.3.18f1
FishNet: 3.11.18R Pro
FishySteamworks: 4.1.0
Discord: https://discord.com/channels/424284635074134018/1248143502253101118

Sometimes we get this exception in build and it's broke gameplay.
Log from host.

```
Server received an unhandled PacketId of 52924 on channel Reliable from connectionId 0. Connection will be kicked immediately.
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
FishNet.Managing.Logging.LevelLoggingConfiguration:LogError(String)
FishNet.Managing.Server.ServerManager:ParseReceived(ServerReceivedDataArgs)
FishNet.Managing.Server.ServerManager:Transport_OnServerReceivedData(ServerReceivedDataArgs)
FishNet.Transporting.Tugboat.Tugboat:HandleServerReceivedDataArgs(ServerReceivedDataArgs)
FishySteamworks.Server.ServerSocket:IterateIncoming()
FishNet.Managing.Transporting.TransportManager:IterateIncoming(Boolean)
FishNet.Managing.Timing.TimeManager:IncreaseTick()
FishNet.Managing.Timing.TimeManager:TickUpdate()

Steam connection not found for connectionId 0.
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
FishySteamworks.FishySteamworks:StopConnection(Int32, Boolean)
FishNet.Managing.Server.ServerManager:Kick(Int32, KickReason, LoggingType, String)
FishNet.Managing.Server.ServerManager:ParseReceived(ServerReceivedDataArgs)
FishNet.Managing.Server.ServerManager:Transport_OnServerReceivedData(ServerReceivedDataArgs)
FishNet.Transporting.Tugboat.Tugboat:HandleServerReceivedDataArgs(ServerReceivedDataArgs)
FishySteamworks.Server.ServerSocket:IterateIncoming()
FishNet.Managing.Transporting.TransportManager:IterateIncoming(Boolean)
FishNet.Managing.Timing.TimeManager:IncreaseTick()
FishNet.Managing.Timing.TimeManager:TickUpdate()

ConnectionId 0 not found within Clients. Connection will be kicked immediately.
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
FishNet.Managing.Logging.LevelLoggingConfiguration:LogError(String)
FishNet.Managing.Server.ServerManager:Kick(Int32, KickReason, LoggingType, String)
FishNet.Managing.Server.ServerManager:ParseReceived(ServerReceivedDataArgs)
FishNet.Managing.Server.ServerManager:Transport_OnServerReceivedData(ServerReceivedDataArgs)
FishNet.Transporting.Tugboat.Tugboat:HandleServerReceivedDataArgs(ServerReceivedDataArgs)
FishySteamworks.Server.ServerSocket:IterateIncoming()
FishNet.Managing.Transporting.TransportManager:IterateIncoming(Boolean)
FishNet.Managing.Timing.TimeManager:IncreaseTick()
FishNet.Managing.Timing.TimeManager:TickUpdate()

Uploading Crash Report
InvalidOperationException: Collection was modified; enumeration operation may not execute.
  at System.Collections.Generic.Dictionary`2+Enumerator[TKey,TValue].MoveNext () [0x00000] in <00000000000000000000000000000000>:0 
  at FishySteamworks.Server.ServerSocket.IterateIncoming () [0x00000] in <00000000000000000000000000000000>:0 
  at FishNet.Managing.Transporting.TransportManager.IterateIncoming (System.Boolean server) [0x00000] in <00000000000000000000000000000000>:0 
  at FishNet.Managing.Timing.TimeManager.IncreaseTick () [0x00000] in <00000000000000000000000000000000>:0 
  at FishNet.Managing.Timing.TimeManager.TickUpdate () [0x00000] in <00000000000000000000000000000000>:0 
```

In similar places in [FizzySteamworks](https://github.com/Chykary/FizzySteamworks/blob/master/com.mirror.steamworks.net/NextServer.cs), connections are copied to avoid this error.

<img width="764" alt="image" src="https://github.com/FirstGearGames/FishySteamworks/assets/430897/c96c4892-3db4-4ad5-b122-9b162b5446d6">

**Comment by f2069** [comment]
Just in case. @FirstGearGames

**Comment by FirstGearGames** [comment]
Thought I had replied to this awhile ago. This will get looked at as a priority given it's potential breaking capabilities.

**Comment by petergiuntoli** [comment]
We're seeing a similar issue in fishnet 4.3.1 with Steamworks 4.1.0

```
InvalidOperationException: Collection was modified; enumeration operation may not execute.
  at System.Collections.Generic.Dictionary`2+Enumerator[TKey,TValue].MoveNext () [0x00013] in <eef08f56e2e042f1b3027eca477293d9>:0 
  at FishySteamworks.Server.ServerSocket.IterateIncoming () [0x000f4] rog\Assets\Plugins\FishNet\Plugins\FishySteamworks\Core\ServerSocket.cs:328 
  at FishySteamworks.FishySteamworks.IterateIncoming (System.Boolean server) [0x00003] rog\Assets\Plugins\FishNet\Plugins\FishySteamworks\FishySteamworks.cs:252 
  at FishNet.Transporting.Multipass.Multipass.IterateIncoming (System.Boolean server) [0x00015] in rog\Assets\Plugins\FishNet\Runtime\Transporting\Transports\Multipass\Multipass.cs:428 
  at FishNet.Managing.Transporting.TransportManager.IterateIncoming (System.Boolean server) [0x00012] in rog\Assets\Plugins\FishNet\Runtime\Managing\Transporting\TransportManager.cs:715 
  at FishNet.Managing.Timing.TimeManager.TryIterateData (System.Boolean incoming) [0x0001a] in rog\Assets\Plugins\FishNet\Runtime\Managing\Timing\TimeManager.cs:1029 
  at FishNet.Managing.Timing.TimeManager.IncreaseTick () [0x000f1] in rog\Assets\Plugins\FishNet\Runtime\Managing\Timing\TimeManager.cs:683 
  at FishNet.Managing.Timing.TimeManager.<TickUpdate>g__MethodLogic|97_0 () [0x00000] in rog\Assets\Plugins\FishNet\Runtime\Managing\Timing\TimeManager.cs:344 
  at FishNet.Managing.Timing.TimeManager.TickUpdate () [0x0005a] in rog\Assets\Plugins\FishNet\Runtime\Managing\Timing\TimeManager.cs:334 
  at FishNet.Transporting.NetworkReaderLoop.Update () [0x00000] in rog\Assets\Plugins\FishNet\Runtime\Transporting\NetworkReaderLoop.cs:28 
  ```

**Comment by FirstGearGames** [comment]
Yes it's a transport bug. It will be fixed soon.

**Comment by FirstGearGames** [comment]
Resolved in 4.1.1
